File: discover_ruby/manipulate_the_data.rb
require 'httpclient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 576bcef0e6a9e47b7bbb47385326d42a630bfff6'
}

clnt = HTTPClient.new

res = clnt.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', query = nil, extheader = {})

json_parse = JSON.parse(res)
array = json_parse["items"]
join_array = array.map {|a| a["full_name"]}
puts join_array.join("\n")

File: discover_more_python/manipulate_the_data.py
#!/usr/bin/python

import urllib2
import json

request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token 42ee520bd99c8951da18894d2e0cd0c06d957089'
}

def printResults(data):
    theJSON = json.loads(data)
    for item in theJSON["items"]:
        print item["full_name"]

def main():
    gitData = "https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc"
    webUrl = urllib2.urlopen(gitData)
    if (webUrl.getcode() == 200):
        data = webUrl.read()
        printResults(data)
    else:
        print "uh oh!"

if __name__ == "__main__":
  main()
